src/dbus.c

```diff
@@ -163,13 +166,15 @@ DBUS_METHOD(dunst_NotificationAction);
 DBUS_METHOD(dunst_NotificationCloseAll);
 DBUS_METHOD(dunst_NotificationCloseLast);
 DBUS_METHOD(dunst_NotificationShow);
+DBUS_METHOD(dunst_NotificationListAll);
 DBUS_METHOD(dunst_Ping);
 static struct dbus_method methods_dunst[] = {
         {"ContextMenuCall",        dbus_cb_dunst_ContextMenuCall},
         {"NotificationAction",     dbus_cb_dunst_NotificationAction},
         {"NotificationCloseAll",   dbus_cb_dunst_NotificationCloseAll},
         {"NotificationCloseLast",  dbus_cb_dunst_NotificationCloseLast},
         {"NotificationShow",       dbus_cb_dunst_NotificationShow},
+        {"NotificationListAll",    dbus_cb_dunst_NotificationListAll},
```

Comment:
`NotificationListAll` is currently not a good name, since it only lists the notifications from history. You could include all notifications and add a history attribute or rename the function to something like `NotificationListHistory`.
What are you planning on using this for actually?

Reply:
Basically reviewing notifications that expired but weren't shown (or at least seen) by the user.
![image](https://user-images.githubusercontent.com/4696350/141172359-156c0c90-5e66-4633-bb26-a6b18a4ee05f.png)
ie. scripting some way to show notifications of the past, like a sort-of notification history

Reply:
```result=$(dunstctl history | jq -r '.data[][].summary.data'|rofi -dmenu -i)```

Reply:
and i agree the current name is kinda bad, so ill fix it for now

Reply:
on top of that i could implement so that it also prints the actions (in the same dbus data transfer), allowing the user to implement their own handler for old events

Reply:
> on top of that i could implement so that it also prints the actions (in the same dbus data transfer), allowing the user to implement their own handler for old events

I don't think that's possible after the notification is closed.

Reply:
> Basically reviewing notifications that expired but weren't shown (or at least seen) by the user.
> ![image](https://user-images.githubusercontent.com/4696350/141172359-156c0c90-5e66-4633-bb26-a6b18a4ee05f.png)
> ie. scripting some way to show notifications of the past, like a sort-of notification history

Cool. Wouldn't a "history pop id" be useful for that as well? That way you can take a look at an old notification again.

Reply:
yep, that would be a neat idea

Reply:
also should that be for all notifications or just the ones from history?

Reply:
> also should that be for all notifications or just the ones from history? i assume the latter because of the name

Yes, the latter. We already have history-pop, but history-pop-id would bring back a specific notification from history. I already took a stab at implementing it, but we decided it wouldn't really be useful in many situations (see https://github.com/dunst-project/dunst/pull/823).

Reply:
@fwsmit also done